target/ppc/translate/vsx-impl.c.inc

```diff
@@ -2124,6 +2124,88 @@ static void gen_xvxsigdp(DisasContext *ctx)
     tcg_temp_free_i64(xbl);
 }
 
+static bool do_lstxv(DisasContext *ctx, arg_D *a, bool store, bool paired)
+{
+    TCGv ea;
+    TCGv_i64 xt;
+    MemOp mop;
+    int offset;
+
+    if (paired) {
+        REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    }
+
+    REQUIRE_VSX(ctx);
+
+    ea = tcg_temp_new();
+
+    mop = DEF_MEMOP(MO_Q);
+
+    gen_set_access_type(ctx, ACCESS_INT);
+
+    xt = tcg_temp_new_i64();
+
+    if (ctx->le_mode) {
+        do_ea_calc(ctx, a->ra, tcg_constant_tl(a->si + (paired ? 24 : 8)), ea);
+        offset = -8;
+    } else {
+        do_ea_calc(ctx, a->ra, tcg_constant_tl(a->si), ea);
+        offset = 8;
+    }
+
```

Comment:
Not 100% sure, but I think we need to handle the multithreaded case. For simplicity, I think we can do this without a helper for now, so it would be something like:
```suggestion

    if (tb_cflags(ctx->base.tb) & CF_PARALLEL) {
        gen_helper_exit_atomic(cpu_env);
        ctx->base.is_jmp = DISAS_NORETURN;
        return true;
    }

```
To test this, you will need to run `qemu-system-ppc64` with at least `-smp 2` and maybe `-accel tcg,thread=multi`

Reply:
Oh, I was wrong. PowerISA only requires atomic quadword operations for lq, stq, lqarx and stqcx.. So, nothing to see here :)